crates/sui-rpc/src/client/v2/coin_selection.rs

```diff
@@ -0,0 +1,122 @@
+use futures::StreamExt;
+use prost_types::FieldMask;
+
+use crate::client::v2::Client;
+use crate::client::v2::Result;
+use crate::field::FieldMaskUtil;
+use crate::proto::sui::rpc::v2::ListOwnedObjectsRequest;
+use crate::proto::sui::rpc::v2::Object;
+
+pub const SUI_COIN_TYPE: &str = "0x2::sui::SUI";
+
+impl Client {
+    /// Selects coins of a specific type owned by an address until the total value meets the required amount.
+    ///
+    /// # Arguments
+    /// * `owner_address` - The address that owns the coins
+    /// * `coin_type` - The type of coins to select (e.g., "0x2::sui::SUI")
+    /// * `amount` - The minimum total amount needed
+    /// * `exclude` - Array of object IDs to exclude from selection
+    ///
+    /// # Returns
+    /// A vector of `Object` instances representing the selected coins
+    ///
+    /// # Errors
+    /// Returns an error if there are insufficient funds to meet the required amount or if there is an RPC error
+    pub async fn select_coins(
+        &self,
+        owner_address: impl Into<String>,
+        coin_type: impl Into<String>,
+        amount: impl Into<u128>,
```

Comment:
I'd probably just make this a u64 since thats the only reasonable balance that we can have.